### DISCUSSION TITLE
Feed JSON response to chained request

### DISCUSSION THREAD

**Discussion by rangolucas** [postmanlabs/postman-flows#212] [Q&A] [answered]
I am trying to find a way to implement this flow which I thought would be pretty straightforward:
I have two requests:

- a GET request whose response body is an array of JSON objects
- a POST request that needs to be passed these objects one by one in its request body

is there a way to send the first request and then trigger the second request for each of the objects in the response?

EDIT: as I've seen in a recent [discussion](https://github.com/postmanlabs/postman-flows/discussions/176), there might be an issue with the "For Each" block, which could mean there is no way to do this as of now. If this is the case, I would like to know if it is at least possible to send the whole response (the full array) as-is through the subsequent request.

**Comment by saswatds** [comment, marked as answer]
Hey @rangolucas, you should be able use the `For Each` block. If you have an example defined and flows recognizes your array as a list then `For Each` block will emit each of the objects as different packets and you send request will be triggered. 
 
The various issues in previous discussion was around concatenation and loosing of types on refresh. If you have defined an Example and are not dealing with complicated object concatenation, the `For Each` block will work just fine. Hope that helps

**Reply by rangolucas** [reply]
Thanks for the answer! I was having issues with type inference using the `For Each` block with Postman client but that seems to work on the web: now I can successfully iterate the array in the response as expected.
However, I am still struggling to implement this flow. Should the `For Each` block pass objects in the response directly to the next request payload? That is how I thought it would work, but apparently the subsequent requests are being sent with an empty body.

**Reply by saswatds** [reply]
Understood. So the way flows works is that it just makes the information available to Request in the [Data Variable Scope](https://learning.postman.com/docs/sending-requests/variables/#choosing-variable-scope), but it does not inject anything into the request. For injection, you need to use the `{{variable}}` templating to inject the data.
I will give an example. 
Let's say this is the object of each of your elements in the list
<img width="238" alt="Screenshot 2022-01-19 at 9 13 25 AM" src="https://user-images.githubusercontent.com/8796447/150060148-0b7670d8-8bf3-44f3-bb1d-c7386cacff08.png">

And you have fed the data to the Send Request block like this 
<img width="589" alt="Screenshot 2022-01-19 at 9 16 32 AM" src="https://user-images.githubusercontent.com/8796447/150061133-97243a79-723c-4be6-aeb2-5a33cdaf9319.png">

There are two cases- 
1. You are creating a new request
2. You have variables already defined in the request

## 1. Creating a new request
You can do something like this (as all data provided to the variables ports are flattened)
<img width="556" alt="Screenshot 2022-01-19 at 9 31 55 AM" src="https://user-images.githubusercontent.com/8796447/150061812-3f92af5b-a5f0-45ee-8b7f-71b6044b33ef.png">

## 2. You have variables already defined in the request
<img width="670" alt="Screenshot 2022-01-19 at 9 35 09 AM" src="https://user-images.githubusercontent.com/8796447/150062052-f06b5c41-2234-4faa-a636-0673c20b8e04.png">
<img width="801" alt="Screenshot 2022-01-19 at 9 47 51 AM" src="https://user-images.githubusercontent.com/8796447/150063307-e307a1f4-99df-46e7-91b2-2ec9e456ae5f.png">

Let me know if this helpful.

**Reply by rangolucas** [reply]
That's very useful. Thanks!
